### PR TITLE
#16: fixes mailer and test

### PR DESCRIPTION
**Pull request by kimong1**
Fixes mailer and test

### CODE REVIEW

File: services/mailer.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+const logger = {
+    info: jest.fn(),
+    error: jest.fn()
+}
+logGen.mockReturnValue(logger)
+
+jest.mock('mailgun-js')
+const mailgun = require('mailgun-js');
+const messages = {
+    send: jest.fn().mockImplementation((data) => {
```
Comment (songz) [inline services/mailer.test.js:12]: I think a better way is to us pointers.
```
const messages = {
  send: jest.fn().mockReturnValue(Promise.resolve('hello'))
}
mailgun.mockImplementation( () => {
  return {messages}
})
```

File: services/mailer.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    it('should test if mocksend and mailgun is called', async () => {
+        await email.sendConfirmationEmail('paul@github.com', 'token123')
+        expect(messages.send).toHaveBeenCalledTimes(1)
+        expect(messages.send.mock.calls[0][0]).toMatchSnapshot()
+        expect(logger.info).toHaveBeenCalledTimes(1)
+        expect(logger.info.mock.calls[0][0]).toEqual('Confirmation Email successfully sent')
+    })
+
+    it('should call logger.error when function is called with invalid argument', async () => {
+        await email.sendConfirmationEmail(null, null)
```
Comment (songz) [inline services/mailer.test.js:40]: you should mock reject, I think its important to test here when service is down. like:
```
messages.send = jest.fn().mockReturnValue( Promise.reject('invalid') )
```

File: services/mailer.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+})
+const email = require('./mailer');
+
+describe('Test mailgun', ()=>{
+    beforeEach(() => {
+        jest.clearAllMocks()
+    })
+
+    it('should test if mocksend and mailgun is called', async () => {
+        await email.sendConfirmationEmail('paul@github.com', 'token123')
```
Comment (songz) [inline services/mailer.test.js:32]: read comment below.
```
messages.send = jest.fn().mockReturnValue( Promise.resolve('hello') )
```

File: services/mailer.test.js (hunk 1 above)
Comment (kimong1) [reply services/mailer.test.js:12]: What is pointers?

Comment (songz) [reply services/mailer.test.js:12]: non-primitive data types?

File: services/mailer.test.js (hunk 2 above)
Comment (kimong1) [reply services/mailer.test.js:40]: I think it does the same as what it does in 13-14 lines. I see how it's easier to be read. Is that what you meant?

**Pull request merged**